src/lxc/console.c

```diff
@@ -204,6 +204,152 @@ void lxc_console_signal_fini(struct lxc_tty_state *ts)
 	free(ts);
 }
 
+static int lxc_console_truncate_log_file(struct lxc_console *console)
+{
+	/* be very certain things are kosher */
+	if (!console->log_path || console->log_fd < 0)
+		return -EBADF;
+
+	return lxc_unpriv(ftruncate(console->log_fd, 0));
+}
+
+static int lxc_console_rotate_log_file(struct lxc_console *console)
+{
+	int ret;
+	size_t len;
+	char *tmp;
+
+	/* rotate the console log file */
+	if (!console->log_path || console->log_rotate == 0)
+		return -EOPNOTSUPP;
+
+	/* be very certain things are kosher */
+	if (console->log_fd < 0)
+		return -EBADF;
+
+	len = strlen(console->log_path) + sizeof(".1");
+	tmp = alloca(len);
+
+	ret = snprintf(tmp, len, "%s.1", console->log_path);
+	if (ret < 0 || (size_t)ret >= len)
+		return -EFBIG;
+
+	close(console->log_fd);
+	console->log_fd = -1;
+	ret = lxc_unpriv(rename(console->log_path, tmp));
+	if (ret < 0)
+		return ret;
+
+	return lxc_console_create_log_file(console);
+}
+
+static int lxc_console_write_log_file(struct lxc_console *console, char *buf,
+				      int bytes_read)
+{
+	int ret;
+	int64_t space_left = -1;
+	struct stat st;
+
+	if (console->log_fd < 0)
+		return 0;
+
+	/* A log size <= 0 means that there's no limit on the size of the log
+         * file at which point we simply ignore whether the log is supposed to
+	 * be rotated or not.
+	 */
+	if (console->log_size <= 0)
+		return lxc_write_nointr(console->log_fd, buf, bytes_read);
+
+	/* Get current size of the log file. */
+	ret = fstat(console->log_fd, &st);
+	if (ret < 0) {
+		SYSERROR("Failed to stat the console log file descriptor");
+		return -1;
+	}
+
+	/* handle non-regular files */
+	if ((st.st_mode & S_IFMT) != S_IFREG) {
+		/* This isn't a regular file. so rotating the file seems a
+		 * dangerous thing to do, size limits are also very
+		 * questionable. Let's not risk anything and tell the user that
+		 * he's requesting us to do weird stuff.
+		 */
+		if (console->log_rotate > 0 || console->log_size > 0)
+			return -EINVAL;
+
+		/* I mean, sure log wherever you want to. */
+		return lxc_write_nointr(console->log_fd, buf, bytes_read);
+	}
+
+	/* check how much space we have left */
```

Comment:
The useful variable name makes the comment unnecessary.

Reply:
removed